### PR TITLE
:book: Fix Directory name with space in it AND update 

### LINKED ISSUE #2504: bug: "Generate and Push Docs" action appears to fail on fork branches named "main"
### Describe the bug

This is a problem which I've observed on several occasions, where the the action to create (preview) webpages from a fork of the repository appears to fail if the branch of that fork is "main"
If a branch of that fork is created named something else, the action works.

### Steps To Reproduce

1. Create a fork of kubestellar main in your own account.
2. Go to that fork. (switch to your copy of the repo in github.com)
3. Leave the branch in your copy of the repo named main.
4. Go to Actions and select **Generate and Push Docs**
5. Click the **Run Workflow** button. Select the main branch and click the second **Run Workflow** button in the dialog
6. Workflow runs with errors, and the preview site (https://youraccount.github.io/kubestellar/main) is not generated

### Expected Behavior

The webpages should be generated and the preview website should be available.

### Want to contribute?

- [ ] I would like to work on this issue.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by KPRoche**
## This PR does two things:
1. it renames the website content folder "Contribution Guidelines" to "contribution-guidelines" to yield URLs more compliant with good webdesign practices
2. It adds information about rendering a publicly viewable preview site:
    1. Adds instructions to the /docs/README.md file (which is included into a file in the Contributing section of the website) on how to render a publicly viewable preview site from a fork of the repository.
    2. Adds a reference link to that section from the "Testing website PRs" page
   
There were numerous commits while getting the format cleaned up; they have been squashed into a much smaller set of commits.

Preview of the site is at https://kproche.github.io/kubestellar/doc-addweb
The most modified page is at 
https://kproche.github.io/kubestellar/doc-addweb/contribution-guidelines/operations/document-management/ 
which is generated from https://github.com/KPRoche/kubestellar/blob/doc-addweb/docs/README.md

[note; normally I would have done this as two PRs but the space (%20) character in the URLs was increasingly frustrating to deal with in test renders]
## Related issue(s)

Fixes #2504

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -65,6 +70,49 @@ make serve-docs
...
+-----
+
+### Serving up documents globally from a fork of the repository via GitHub
+You can also take advantage of the "Generate and Push Docs" Action via the github web interface to create an online, shareable rendering of the website.
+This is particularly useful for documentation PRs, as it allows you to share a preview of your proposed changes directly via a URL to a working website.
+To take advantage of this action, you must ensure that you have forked the repository properly, so your fork includes the gh-pages branch required for the Action to run properly:
+
+#### Creating a fork that can use the Generate and Push Docs Action
+
+ 1. Log into your GitHub account via webbrowser
```
Comment (MikeSpreitzer) [inline docs/README.md:82]: Separate 'webbrowser' into two words.

File: docs/README.md [hunk 2]
```diff
@@ -65,6 +70,49 @@ make serve-docs
...
+ 2. Navigate to github.com/kubestellar/kubestellar
+ 3. Select the **Forks** dropdown and click on the plus sign to create a new fork ![image](https://github.com/user-attachments/assets/c0b56897-c2c4-479f-ba16-c1edd4690946)
+ 4. In the resulting dialog select your account as the owner, pick a repository name for the fork, and _be sure to uncheck the_ **"copy the** main **branch only"** _box_ <br />
+![image](https://github.com/user-attachments/assets/c5909ddd-3bf6-44c2-9102-c07c7e1d6a05)
+
+#### If you already created a fork but only included the main branch
+You can remedy the problem by propagating the _gh-pages_ branch into your fork using _git_ commands
```
Comment (MikeSpreitzer) [inline docs/README.md:89]: I would set off "git" with code quotes rather than emphasis.

**Comment by MikeSpreitzer** [comment]
You should make a practice of including a pointer to the preview in your opening comment. In this case it is https://kproche.github.io/kubestellar/doc-addweb/readme/

File: docs/README.md [hunk 3]
```diff
@@ -65,6 +70,49 @@ make serve-docs
...
+ 
+#### Generating a website rendered from a branch of your fork
+1. Work on the documents in a branch of your fork of the repository, and commit the changes
+2. (If you have been working on a local copy of the files, push the changes to the fork, then log into the GitHub webpage for your fork)
+3. Switch to the Actions tab in the top menu bar of the repository page
+4. Select Generate and Push Docs from the list of Actions on the left
+5. Click on the Run Workflow button on the right ![image](https://github.com/user-attachments/assets/5d3d23be-6c8f-454e-bf2c-0c58c8894957)
+6. Select the branch you wish to render and click on the second Run Workflow Button ![image](https://github.com/user-attachments/assets/427b827d-555c-4d36-b9c8-485eda002428)
+7. If that workflow completes successfully, it will automatically call the **Pages build and deployment** workflow.
+8. After a minute or so, you should be able to preview your new version of the website at `https://${repo_owner}.github.io/${fork_name}/${branch_name}`
```
Comment (MikeSpreitzer) [inline docs/README.md:99]: I would describe, even if only in one brief outline-type sentence, how to know when the "Pages build and deployment" workflow finishes.

File: docs/README.md [hunk 4]
```diff
@@ -65,6 +70,49 @@ make serve-docs
...
+
+#### Automatically generate webpages
+If you create a branch of your fork that begins with **doc-** (e.g. _doc-myversion_) the workflow will trigger automatically when you commit changes to the branch.
```
Comment (MikeSpreitzer) [inline docs/README.md:102]: Misplaed modifier here: "that begins with..." follows "fork" but you mean to modify the branch. Also: you mean to modify _the name of_ the branch.

**Comment by MikeSpreitzer** [comment]
Somehow this PR has acquired its own copy of two of my commits that are already in `main`.

**Comment by KPRoche** [comment]
> You should make a practice of including a pointer to the preview in your opening comment. In this case it is https://kproche.github.io/kubestellar/doc-addweb/readme/

Well, bother. I had one the first time I started this PR, then realized I should do the squash and forgot to include it when I actually pushed the Create a PR button.

**Comment by KPRoche** [comment]
> Somehow this PR has acquired its own copy of two of my commits that are already in `main`.

That would be me making a mistake in my squash.

File: docs/README.md (hunk 1 above)
Comment (KPRoche) [reply docs/README.md:82]: Quite right. Fixing

File: docs/README.md (hunk 3 above)
Comment (KPRoche) [reply docs/README.md:99]: Added, with a screenclip

File: docs/README.md (hunk 4 above)
Comment (KPRoche) [reply docs/README.md:102]: expanded the sentence to make it clearer

File: docs/README.md (hunk 3 above)
Comment (MikeSpreitzer) [reply docs/README.md]: Looks good.

FYI, if you wanted to, you could go further and mention that the link to the preview can be found on the page for the **pages build and deployment** workflow.

image

**Comment by MikeSpreitzer** [comment]
There are still copies of two merged commits here.
You could edit a reference to the preview into the opening comment.

**Comment by KPRoche** [comment]
> There are still copies of two merged commits here. You could edit a reference to the preview into the opening comment.

I've edited the opening comment. How do I get rid of the copied merged commits?

**Comment by MikeSpreitzer** [comment]
> How do I get rid of the copied merged commits?

Assuming you have a "remote" named `upstream` that refers to the shared repo:

```shell
git checkout doc-addweb
git fetch upstream
git rebase upstream/main
```

and, assuming no conflicts or you have merged the conflicts and completed the rebase, and also assuming that you have a "remote" named `origin` that refers to your fork of the kubestellar repo:

```shell
git push -f origin doc-addweb
```

**Comment by KPRoche** [comment]
> > How do I get rid of the copied merged commits?
> 
> Assuming you have a "remote" named `upstream` that refers to the shared repo:
> 
> ```shell
> git checkout doc-addweb
> git fetch upstream
> git rebase upstream/main
> ```
> 
> and, assuming no conflicts or you have merged the conflicts and completed the rebase, and also assuming that you have a "remote" named `origin` that refers to your fork of the kubestellar repo:
> 
> ```shell
> git push -f origin doc-addweb
> ```

Done! Thanks!

**Comment by KPRoche** [comment]
Bother, that appears to have lost my most recent edits to the text. Fortunately I saved a backup zipfile, making it easy to restore them.

**Comment by MikeSpreitzer** [comment]
You can also make a backup at any time of your current branch using `git checkout -B backup`.

**Review by MikeSpreitzer — APPROVED** [review]
/lgtm
/approve
Thanks!

**Pull request merged**